### PR TITLE
usage rpc methods

### PR DESCRIPTION
**Pull request by furkhat**

### CODE REVIEW

File: doc/ui/rpc.md [hunk 1]
```diff
@@ -96,6 +98,48 @@ This document describes a UI JSON RPC API located in "ui" namespace.
...
 
+#### Usage At Channel
+
+*Method*: `usageAtChannel`
+
+*Description*: Returns total units used for a given channel.
+
+*Parameters*: 
+1. Password (string)
+2. Channel (string)
```
Comment (ababo) [inline doc/ui/rpc.md]: Channel id (string)

File: doc/ui/rpc.md [hunk 2]
```diff
@@ -96,6 +98,48 @@ This document describes a UI JSON RPC API located in "ui" namespace.
...
+
+#### Usage At Offering
+
+*Method*: `usageAtOffering`
+
+*Description*: Returns total units used for all channels with a given offering.
+
+*Parameters*: 
+1. Password (string)
+2. Offering (string)
```
Comment (ababo) [inline doc/ui/rpc.md]: Offering id (string)

File: doc/ui/rpc.md [hunk 3]
```diff
@@ -96,6 +98,48 @@ This document describes a UI JSON RPC API located in "ui" namespace.
...
+1. Password (string)
+2. Channel (string)
+
+*Result*:
+1. Amount (number)
+
+
+#### Usage At Offering
+
+*Method*: `usageAtOffering`
```
Comment (ababo) [inline doc/ui/rpc.md]: Why not `GetOfferingUsage`?

File: doc/ui/rpc.md [hunk 4]
```diff
@@ -96,6 +98,48 @@ This document describes a UI JSON RPC API located in "ui" namespace.
...
+
+*Parameters*: 
+1. Password (string)
+2. Offering (string)
+
+*Result*:
+1. Amount (number)
+
+
+#### Usage At Product
```
Comment (ababo) [inline doc/ui/rpc.md]: Why not `GetProductUsage`?

File: doc/ui/rpc.md [hunk 5]
```diff
@@ -96,6 +98,48 @@ This document describes a UI JSON RPC API located in "ui" namespace.
...
+
+*Method*: `usageAtProduct`
+
+*Description*: Returns total units used in all channel of all offerings with a given product.
+
+*Parameters*: 
+1. Password (string)
+2. Product (string)
```
Comment (ababo) [inline doc/ui/rpc.md]: id

File: ui/usage.go [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+// UsageAtOffering returns total units used for all channels
+// with a given offering.
+func (h *Handler) UsageAtOffering(password, offeringID string) (*UsageResult, error) {
+	logger := h.logger.Add("method", "UsageAtOffering",
+		"offeringID", offeringID)
+
+	return h.usage(logger, password,
+		`SELECT SUM(sessions.units_used)
```
Comment (ababo) [inline ui/usage.go]: Cool SQL formatting!

File: doc/ui/rpc.md [hunk 7]
```diff
@@ -34,6 +34,48 @@ This document describes a UI JSON RPC API located in "ui" namespace.
...
+*Method*: `getChannelUsage`
+
+*Description*: Returns total units used for a given channel.
+
+*Parameters*: 
+1. Password (string)
+2. Channel id (string)
+
+*Result*:
+1. Amount (number)
```
Comment (ababo) [inline doc/ui/rpc.md]: Please remove compound returns, as we agreed.

File: ui/usage.go [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+		   FROM sessions
+		   WHERE channel=$1`, channelID)
+}
+
+// GetOfferingUsage returns total units used for all channels
+// with a given offering.
+func (h *Handler) GetOfferingUsage(password, offeringID string) (uint, error) {
```
Comment (dzeckelev) [inline ui/usage.go]: *uint and in other places

File: ui/usage.go [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+
+// GetChannelUsage returns total units used for a given channel.
+func (h *Handler) GetChannelUsage(password, channelID string) (uint, error) {
+	logger := h.logger.Add("method", "GetChannelUsage",
+		"channelID", channelID)
+
+	return h.usage(logger, password,
+		`SELECT SUM(sessions.units_used)
+		   FROM sessions
+		   WHERE channel=$1`, channelID)
```
Comment (dzeckelev) [inline ui/usage.go]: line shifted

**Pull request merged**